src/scheduler/check.c

```diff
@@ -505,15 +505,15 @@ shrink_to_run_event(status *policy, server_info *sinfo,
 			/* If there are no events left to check or if we have reached the front of event list or
 			 * if the event is falling before min end time, break.
 			 */
-			if ((te == NULL && last_skipped_event == NULL) || te == initial_event || te->event_time < min_end_time)
-				break;
-			/* If no events in this segment, then try last skipped event of the previous segment */
-			else if (te == NULL) {
+			if (te == NULL)
```

Comment:
so we don't wanna break if te == NULL && last_skipped_event == NULL ?

Reply:
Clang complained about the end of that if statement.  If te == NULL last_skipped_event != NULL and te->event_time.  That last part would dereference and crash.  You are right that swapping the order of the two if/else cases is wrong by itself.  That first part would never trigger.  I broke the condition into two parts.  It should work better now.

Reply:
I'm not sure I'm following you. Couldn't you just change the last condition from (te->event_time < min_end_time) to (te != NULL && te->event_time < min_end_time) ? Or do we want to change the behavior so that we won't break out of the loop if (te == NULL && last_skipped_event == NULL) ? Because with your changes that won't happen.

Reply:
I think I fixed this without changing behavior.  I am not too familiar with this function and it is very complex.  I'd rather not make changes that might break things.  I left the if/else flipped, and added a check if last_skipped_event == NULL inside the check for if (te == NULL).   This basically broke the single break in to two.